assemblies/assembly_architect-decision-services/states-con.adoc

```diff
@@ -0,0 +1,345 @@
+----
+KieServices kieServices = KieServices.Factory.get();
+KieContainer kContainer = kieServices.getKieClasspathContainer();
+----
+
+The above code snippet compiles all the DRL files found on the classpath and adds the result of this compilation, a ``KieModule``, in the ``KieContainer``. If there are no errors, we create our session from the `KieContainer` and execute against some data:
+
+[source,java]
+----
+StatelessKieSession kSession = kContainer.newStatelessKieSession();
+Applicant applicant = new Applicant( "Mr John Smith", 16 );
+assertTrue( applicant.isValid() );
+ksession.execute( applicant );
+assertFalse( applicant.isValid() );
+----
+
+The preceding code executes the data against the rules. Since the applicant is under the age of 18, the application is marked as invalid.
+
+So far we've only used a single instance, but what if we want to use more than one, we can execute against any object implementing Iterable interface, such as a collection. We will add another class called `Application`, which has the date of the application, and we will move the `boolean valid` field to the `Application` class.
+
+[source,java]
+----
+public class Applicant {
+    private String name;
+    private int age;
+    // getter and setter methods here
+}
+
+public class Application {
+    private Date dateApplied;
+    private boolean valid;
+    // getter and setter methods here
+}
+----
+
+
+Now, we can add another rule to validate that the application was submitted within a certain period of time.
+
+[source]
+----
+package com.company.license
+
+rule "Is of valid age"
+when
+    Applicant( age < 18 )
+    $a : Application()
+then
+    $a.setValid( false );
+end
+
+rule "Application was made this year"
+when
+    $a : Application( dateApplied > "01-jan-2009" )
+then
+    $a.setValid( false );
+end
+----
+
+As Java arrays do not implement the `Iterable` interface, we use the `Arrays.asList(...)` JDK converter method .
+The code shown below executes against an iterable list, where all collection elements are inserted before any matched rules are fired.
+
+[source,java]
+----
+StatelessKieSession kSession = kContainer.newStatelessKieSession();
+Applicant applicant = new Applicant( "Mr John Smith", 16 );
+Application application = new Application();
+assertTrue( application.isValid() );
+ksession.execute( Arrays.asList( new Object[] { application, applicant } ) );
+assertFalse( application.isValid() );
+----
+
+The `execute(Object object)` and `execute(Iterable objects)` execute methods are convenience methods for the interface ``BatchExecutor``'s method ``execute(Command command)``.
+
+The `KieCommands` commands factory, which is obtainable from the `KieServices` like all other factories of the KIE API, is used to create commands, so that the following is equivalent to ``execute(Iterable it)``:
+
+[source,java]
+----
+ksession.execute( kieServices.getCommands().newInsertElements( Arrays.asList( new Object[] { application, applicant } ) );
```

Comment:
Could we please update as:
- we do not need instantiate array
- to make more readable and differentiate more between rule snippets and java snippets:
```java
ksession.execute(
    kieServices.getCommands()
               .newInsertElements(Arrays.asList(application, applicant)
);
```

Reply:
Done